evm-ext/src/solidity_abi.rs

```diff
@@ -288,8 +292,11 @@ fn sol_to_value(sv: DynSolValue) -> Value {
             256 => Value::u256(u.to_move_u256()),
             _ => unreachable!("Only 8, 16, 32, 64, 128 and 256 bit uints are supported by move"),
         },
-        // Move doesn't have fixed-length arrays, the output will be the same regardless of the length
-        DynSolValue::FixedBytes(w, _size) => Value::vector_u8(w.to_vec()),
+        // Packing it back into type-erased `SolidityFixedBytes`
+        // TODO: any way to pass back the size type parameter?
```

Comment:
No, we can't enforce types here unfortunately. I think it's ok though because we should only reach this point if the ABI-encoded bytes the user gave were successfully deserialized by the alloy library, and so the type parameters the user gave should have to be right.